backend/src/server/routes/v1/integration-router.ts

```diff
@@ -206,6 +206,34 @@ export const registerIntegrationRouter = async (server: FastifyZodProvider) => {
         id: req.params.integrationId
       });
 
+      if (integration.integration === "aws-secret-manager") {
+        // Fetch additional AWS integration details
+        const awsRoleDetails = await server.services.integration.getIntegrationAWSAssumeRoleArn({
+          actorId: req.permission.id,
+          actor: req.permission.type,
+          actorAuthMethod: req.permission.authMethod,
+          actorOrgId: req.permission.orgId,
+          id: req.params.integrationId
+        });
+
+        if (integration.metadata) {
+          integration.metadata = {
+            ...integration.metadata,
+            awsRegion: integration.region,
+            awsIamRole: awsRoleDetails.role
+          };
+        } else {
+          integration.metadata = {
+            awsRegion: integration.region,
+            awsIamRole: awsRoleDetails.role
+          };
+        }
+      } else {
+        integration.metadata = {
+          region: integration.region
+        };
```

Comment:
On line 222 you set `awsRegion`, but on line 233 you are setting `region` in the metadata object. Is this intentional?